File: Rearrange_Array_Elements_by_Sign.py

nums = [3,1,-2,-5,2,-4]

def rearrangeArra(nums):
    ls = [0]*len(nums)
    p=0
    n=1
    for i in nums:
        if i > 0 :
            ls[p]=i
            p+=2
        else:
            ls[n]=i
            n+=2
    return ls

print(rearrangeArra(nums))

File: Rearrange_Array_Elements_by_Sign/Rearrange_Array_Elements_by_Sign.go
package main

import "fmt"

func main() {
	nums := []int{3, 1, -2, -5, 2, -4}
	fmt.Println(rearrangeArra(nums))
}

func rearrangeArra(nums []int) []int {
	mk := make([]int, len(nums))
	p := 0
	n := 1
	for _, v := range nums {
		if v > 0 {

			mk[p] = v
			p += 2
		} else {
			mk[n] = v
			n += 2
		}
	}
	return mk
}
